File: pytools/examples/csharp/server/Protocol/AckScenePlayers.cs
using System.Collections;
using System.Collections.Generic;


public class AckScenePlayers
{
	private List<MsgScenePlayer> _players = new List<MsgScenePlayer>();


	public Packet Encode()
	{
		Packet packet = new Packet();
		ushort players_count = (ushort)this._players.Count;
		packet.WriteUshort(players_count);
		for (ushort i = 0; i < players_count; i++)
		{
			MsgScenePlayer xxx = this._players[i];
			packet.WriteBuffer(xxx.GetBuffer());
		}
		packet.Encode(Msg.P_ACK_SCENE_PLAYERS);
		return packet;
	}


	public List<MsgScenePlayer> players
	{
		get { return this._players; }
		set { this._players = value; }
	}

}

File: pytools/data/data.code.java.server/AckScenePlayers.java
package protocol;


import network.Packet;
import network.PacketUtil;

import java.util.List;
import java.util.ArrayList;


public class AckScenePlayers
{
	private List<Msgsceneplayer> players = new ArrayList<Msgsceneplayer>();


	public byte[] encode()
	{
		Packet packet = new Packet();
		int players_count = players.size();
		packet.writeShort((short)players_count);
		for (int i = 0; i < players_count; i++)
		{
			packet.writeBytes(players.get(i).getBytes());
		}
		return packet.encode((short)Msg.P_ACK_SCENE_PLAYERS);
	}



	public void setPlayers(List<Msgsceneplayer> players)
	{
		this.players = players;
	}

}
